src/main/java/com/eighthlight/fabrial/App.java

```diff
@@ -0,0 +1,47 @@
+package com.eighthlight.fabrial;
+
+import com.eighthlight.fabrial.server.ServerConfig;
+import com.eighthlight.fabrial.server.TcpServer;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class App {
+  static final Logger logger = Logger.getLogger(App.class.getName());
+
+  static TcpServer server;
+
+  public static void start() {
+    server = new TcpServer(new ServerConfig());
+    try {
+      server.start();
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, "Failed to start server", e);
+      System.exit(1);
+    }
+    Runtime.getRuntime().addShutdownHook(new Thread(App::shutdown));
+  }
+
+  public static void run() {
+    while (!server.isClosed()) {
+      // let server run
+    }
+  }
+
+  public static void shutdown() {
+    logger.info("Shutting down");
+    try {
+      server.close();
+    } catch (IOException e) {
+      logger.log(Level.SEVERE, "Failed to close server", e);
+      System.exit(1);
+    }
+  }
+
+  public static void main(String[] args) {
+    start();
+    run();
```

Comment:
I was surprised to see the `main` method calling both `start` and `run`, which sound a bit like synonyms. The "Goodbye" and "Shutting down" logs also feel a bit like the same thing. So might make sense to boil down to a single method call from `main`.

Reply:
Yeah, I wasn't thrilled about how this turned out either. Same w/ the loop below. Will give it some thought and see if I can come up with a more elegant solution.